drivers/gpio/sabertooth_analog_driver.go

```diff
@@ -0,0 +1,294 @@
+
+func (s *sabertoothAnalogDriver) GetDifferential() float64 { return s.differential }
+
+func (s *sabertoothAnalogDriver) write(pin string, value float64) error {
+    if pin != "" {
+        value = math.Min(math.Max(value, 0), 1)
+        value = gobot.Rescale(value, 0, 1, 0, math.MaxUint8)
+        return s.connection.PwmWrite(pin, byte(math.Round(value)))
+    }
+    return nil
+} // END write
+
+func (s *sabertoothAnalogDriver) SetRampRate(rate float64) error {
+    if !s.switches.Bidirectional() { return ErrBidirectionalOperation }
+    if s.a1 == "" { return ErrA1NotSet }
+    rate = math.Min(math.Max(rate, 0), 1)
+    return s.write(s.a1, rate)
+} // END SetRampRate
+
+func (s *sabertoothAnalogDriver) SetMaxPower(ratio float64) error {
+    if !s.switches.Bidirectional() { return ErrBidirectionalOperation }
+    if s.a2 == "" { return ErrA2NotSet }
+    ratio = math.Min(math.Max(ratio, 0), 1)
+    return s.write(s.a2, ratio)
+} // END SetMaxPower
+
+func (s *sabertoothAnalogDriver) Throttle(throttle float64) error {
+    throttle = math.Min(math.Max(throttle, -1), 1)
+    s.throttle = throttle
+    
+    if s.switches.Bidirectional() {
+        if s.switches.Mixed() {
+            // Bidirectional + Mixed
+            return s.write(s.s1, (throttle / 2) + 0.5)
+        } else {
+            // Bidirectional + Independent
+            
+            // Applies ratio and transforms from a range of -1 to 1 into a range of 0 to 1
+            s1Throttle := ((throttle * s.m1Ratio) / 2) + 0.5
+            s2Throttle := ((throttle * s.m2Ratio) / 2) + 0.5
+            
+            if err := s.write(s.s1, s1Throttle); err != nil { return err }
+            return s.write(s.s2, s2Throttle)
+        }
+    } else {
+        if throttle < 0 && (s.a1 == "" || s.a2 == "") {
+            return ErrUnidirectionalCCW
+        }
+        
+        if s.switches.Mixed() {
+            // Unidirectional + Mixed
+            a := 1.0
+            if throttle < 0 { a = 0.0 }
+            
+            if err := s.write(s.a1, a); err != nil { return err }
+            if err := s.write(s.a2, a); err != nil { return err }
+            return s.write(s.s1, math.Abs(throttle))
+        } else {
+            // Unidirectional + Independent
+            m1Throttle := throttle * s.m1Ratio
+            m2Throttle := throttle * s.m2Ratio
+            
+            applyThrottle := func(t float64, sPin, aPin string) error {
+                a := 1.0
+                if t < 0 { a = 0 }
+                
+                if err := s.write(aPin, a); err != nil { return err }
+                return s.write(sPin, math.Abs(t))
+            }
+            
+            if err := applyThrottle(m1Throttle, s.s1, s.a1); err != nil { return err }
+            return applyThrottle(m2Throttle, s.s2, s.a2)
+        }
+    }
+} // END Throttle
+
+func (s *sabertoothAnalogDriver) Differential(angle float64) error {
```

Comment:
This function looks also quite complicate, so my suggestion is to add some unit test.